app/components/utils/government-areas/area-selector/government-area-selector-form.js

```diff
@@ -13,61 +13,42 @@ class DomainSelection {
 }
 
 export default class GovernmentAreaSelectorForm extends Component {
-  /**
-   * Since fields are children of domains, this component only takes fields as arguments, and calculates the required domains internally
-   * @argument availableFields: All fields that will be listed as options to be checked
-   * @argument selectedFields: Which fields should be checked
-   * @argument onSelectFields: Action, takes an array of fields
-   * @argument onUnSelectFields: Action, takes an array of fields
-   */
-
   @tracked domainSelections;
 
-  get availableFields() {
-    return this.args.availableFields || [];
-  }
-
-  get selectedFields() {
-    return this.args.selectedFields || [];
-  }
-
-  get selectedDomains() {
-    return this.args.selectedDomains || [];
-  }
-
   constructor() {
     super(...arguments);
     this.calculateDomainSelections.perform();
   }
 
   @task
   *calculateDomainSelections() {
-    const domainsFromFields = yield Promise.all(this.availableFields.map((field) => field.broader));
-    const uniqueDomains = [...new Set(domainsFromFields)].sortBy('label');
+    const availableDomains = this.args.availableDomains.sortBy('label');
     const domainSelections = [];
-    for (const domain of uniqueDomains) {
-      // Filter logic is applied in 2 steps, such that promises to fetch the domains can be executed using Promise.all
-      // Step 1: create an array of domains, one for each selected field, using the same order as this.selectedFields
-      // Step 2: use the array of step 1 to verify whether the domain fetched for the field is the current domain
-      const selectedFieldsDomains = yield Promise.all(this.selectedFields.map((field) => field.broader));
-      // eslint-disable-next-line no-unused-vars, id-length
-      const selectedFieldsForDomain = this.selectedFields.filter((_, index) => selectedFieldsDomains[index] === domain);
-
-      // Similar filter logic applied in 2 steps as above for the available fields
-      const availableFieldsDomains = yield Promise.all(this.availableFields.map((field) => field.broader));
-      // eslint-disable-next-line no-unused-vars, id-length
-      const availableFieldsForDomain = this.availableFields.filter((_, index) => availableFieldsDomains[index] === domain);
-      const isSelected = this.selectedDomains.includes(domain);
-
-      domainSelections.push(new DomainSelection(domain, isSelected, availableFieldsForDomain, selectedFieldsForDomain));
+    for (const availableDomain of availableDomains) {
+      const isDomainSelected =
+        this.args.selectedDomains.includes(availableDomain);
+      const availableFieldsForDomain = yield availableDomain.narrower;
```

Comment:
Hier maak je de veronderstelling dat de `narrower` van een government-domain altijd een government-field is, wat niet altijd correct is. De government-domains kunnen een hierarchie vormen. 
Omgekeerd weten we wel dat de `broader` van een government-field` steeds een government-domain is.

Ik zou hier de oude logica houden. Die kan wel geoptimaliseerd worden door `selectedFieldsDomains` en `availableFieldsDomains` uit de for-loop te halen. En dan gewoon een extra check op `deprecated` toe te voegen alvorens de `DomainSelection` te pushen op de array.